projects/rocfft/shared/fft_params.h

```diff
@@ -1996,7 +2155,7 @@ class fft_params
 
     // Given a data type and dimensions, fill the buffer, imposing Hermitian symmetry if necessary.
     template <typename Tbuff>
-    inline void compute_input(std::vector<Tbuff>& input)
+    inline void compute_input(std::vector<Tbuff>& input, const bool use_gpu_symmetrizer = true)
```

Comment:
```suggestion
    inline void compute_input(std::vector<Tbuff>& input, const bool use_gpu_symmetrizer = default_symmetrizer<Tbuff>::use_gpu)
```
Unless I'm mistaken, we currently use the host (resp. gpu) symmetrizer for host (resp. gpu) input buffers so we'd need the default value to depend on the template specialization to keep the same behavior.
This suggestion would achieve that with a `default_symmetrizer` struct defined as
```
template <typename Tbuff>
struct default_symmetrizer;

template <typename T>
struct default_symmetrizer<hostbuf_t<T>>
{
    static constexpr bool use_gpu = false;
};

template <typename T>
struct default_symmetrizer<gpubuf_t<T>>
{
    static constexpr bool use_gpu = true;
};
```
Note 1: the same type_trait struct could be used for declaring default values of the `set_input` variants as well and ensure consistency between the default values for `compute_input` and the corresponding `set_input` it intends to use.
Note 2: this suggestion is irrelevant if it's decided to leave this function unchanged in the end.